packages/back-end/src/services/experiments.ts

```diff
@@ -806,6 +810,244 @@ export function toSnapshotApiInterface(
   };
 }
 
+/**
+ * While the `postMetricValidator` can detect the presence of values, it cannot figure out the correctness.
+ * @param payload
+ * @param datasource
+ */
+export function postMetricApiPayloadIsValid(
+  payload: z.infer<typeof postMetricValidator.bodySchema>,
+  datasource: Pick<DataSourceInterface, "type">
+): { valid: true } | { valid: false; error: string } {
+  const { type, sql, sqlBuilder, mixpanel, behavior } = payload;
+
+  // Validate query format: sql, sqlBuilder, mixpanel
+  let queryFormatCount = 0;
+  if (sqlBuilder) {
+    queryFormatCount++;
+  }
+  if (sql) {
+    queryFormatCount++;
+  }
+  if (mixpanel) {
+    queryFormatCount++;
+  }
+  if (queryFormatCount !== 1) {
+    return {
+      valid: false,
+      error: "Can only specify one of: sql, sqlBuilder, mixpanel",
+    };
+  }
+
+  // Validate behavior
+  if (behavior) {
+    if (
+      typeof behavior.riskThresholdDanger !== "undefined" &&
+      behavior.riskThresholdDanger < 0
+    )
+      return {
+        valid: false,
+        error: "riskThresholdDanger must be a non-negative number",
+      };
+
+    if (
+      typeof behavior.riskThresholdSuccess !== "undefined" &&
+      behavior.riskThresholdSuccess < 0
+    )
+      return {
+        valid: false,
+        error: "riskThresholdSuccess must be a non-negative number",
+      };
+  }
```

Comment:
we have this validation in the UI, not sure if you want to add it as well:
```
value.loseRisk < value.winRisk
      ? "The acceptable risk percentage cannot be higher than the too risky percentage"
      : "";
```

Reply:
I've added this validation including tests.